src/parquet/column_reader.cc

```diff
@@ -212,6 +121,9 @@ bool ColumnReader::ReadNewPage() {
       // Read a data page.
       num_buffered_values_ = current_page_header_.data_page_header.num_values;
```

Comment:
I am not sure if you can still use num_buffered_values_ = current_page_header_.data_page_header.num_values; for the vectorized reader.
current_page_header_.data_page_header.num_values counts the NULLs as well, which are actually not present physically in the DATA_PAGE. Inside the decoder (plain-decoder), you loop on this number to populate the buffers. If you have nulls, you shouldnt be looping on this value.